src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs

```diff
@@ -184,9 +212,25 @@ public void Init(HttpApplication application)
             };
         }
 
-        static bool ShouldLogFormData => Logger.IsEnabled(_formDataLoggingLevel) &&
-            (LogPostedFormData == LogPostedFormDataOption.Always ||
-                (LogPostedFormData == LogPostedFormDataOption.OnlyOnError && HttpContext.Current.Response.StatusCode >= 500));
+        static Func<HttpContext, bool> FormLoggingStrategy
```

Comment:
Because the method allocates a new object for each call, `GetFormLoggingStrategy()` might be more indicative of behaviour.